app/controllers/coupons_controller.rb

```diff
@@ -2,17 +2,31 @@
 
 # CouponsController
 class CouponsController < ApplicationController
+  # include Rectify::ControllerHelpers
+
   def create
     coupon = Coupon.find_by(code: coupon_params[:code])
-    return redirect_to cart_path, alert: t('coupon.not_exist') unless coupon
-    return redirect_to cart_path, alert: t('coupon.used') if coupon.order
-    @order = @current_order
-    @order.update(coupon: coupon)
-    redirect_to cart_path, notice: t('coupon.added')
+    use_coupon(coupon)
   end
 
   private
 
+  def use_coupon(coupon)
+    UseCoupon.call(coupon) do
```

Comment:
Command (service object) should contain the logic, not the controller. Move `update_order_with_coupon` to the command. Finding coupon can be moved to the command as well. You can broadcast updated order from command if needed.
There is no need for `use_coupon` method. It's code can be placed in `create`.